### PR TITLE
Models extending

### LINKED ISSUE #12: Extend and improve existing models
# Feature Request

## Why are you interested in this functionality?

We need to store more useful data in models like "Projects". Also it is important to set attributes in models fields in a proper way: for example names of tags need to be unique so there would not be duplicates in database.

## Additional context or related issues

For project model: 
https://docs.google.com/document/d/1oeohHlzfjGhO0pvQmX24E7yO42DfbFcLnrCgejVBtHk/edit?usp=sharing
For user model:
https://docs.google.com/document/d/15oknkiSFvQI2hq8Cn0pYdsysplj0v6dW5udtBKunLkE/edit?usp=sharing
Docs reference:
https://docs.djangoproject.com/en/2.2/topics/db/models/

### PR DESCRIPTION
**Pull request by nikita03565**
# Pull Request

## Closed issues

Closes #12, #8

### CODE REVIEW

File: educations/serializers.py [hunk 1]
```diff
@@ -24,3 +24,4 @@ class EducationSerializer(serializers.ModelSerializer):
     class Meta:
         model = Education
         fields = "__all__"
+        depth = 1
```
Comment (virrius) [inline educations/serializers.py]: Why do you add the depth parameter only here? My inner perfectionist is worried

File: projects/models.py [hunk 2]
```diff
@@ -47,12 +47,18 @@ class Project(models.Model):
     skills = models.ManyToManyField(Skill, related_name='projects', help_text="Skills required", )
     tags = models.ManyToManyField(Tag, related_name='projects', help_text="Tags related", )
 
-    class Meta:
-        ordering = ('date_start', )
+    def get_project_teachers(self):
```
[the remark this answers is not in the document]
Comment (nikita03565) [reply projects/models.py]: Fixed. I've moved this methods from models and added call of "from projects.services import get_project_teachers, get_project_students" in serializer class to display teachers and students separately

File: users/models.py [hunk 3]
```diff
@@ -56,12 +58,18 @@ class User(AbstractUser):
     )
     username = models.CharField(max_length=255, unique=False)
     bio = models.CharField(max_length=160, null=True, blank=True)
-    middle_name = models.CharField(max_length=30, blank=True)
+    middle_name = models.CharField(max_length=30, null=True, blank=True)
     education = models.ManyToManyField(Education, related_name="users", blank=True, )
     skills = models.ManyToManyField(Skill, related_name="Users", help_text="Skills possessed by the user", blank=True, )
+    phone_regex = RegexValidator(regex=r'^\+?1?\d{9,15}$', message="Phone number must be entered in the format:"
+                                                                   " '+999999999'. Up to 15 digits allowed.")
```
[the remark this answers is not in the document]
Comment (nikita03565) [reply users/models.py]: Fixed. Set only 11 digits

File: users/models.py [hunk 4]
```diff
@@ -56,12 +58,18 @@ class User(AbstractUser):
...
+    phone_number = models.CharField(validators=[phone_regex], max_length=17, null=True, blank=True)
+    date_of_birth = models.DateField("Date of birth", null=True, blank=True,
```
[the remark this answers is not in the document]
Comment (nikita03565) [reply users/models.py]: We need to discuss this more. Also I suggest that we may need to add is_student flag in order to render different fields and do some other stuff for these different types of users

File: educations/serializers.py (hunk 1 above)
Comment (nikita03565) [reply educations/serializers.py]: Explained this already

**Pull request merged**